### DISCUSSION TITLE
Gerrit only: Renovate doesn't properly allow specifying commit trailers through the `commitBody` config

### DISCUSSION THREAD

**Discussion by travisshivers** [renovatebot/renovate#33306] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gerrit, 39.82.3

### Please tell us more about your question or problem

The [commitBody](https://docs.renovatebot.com/configuration-options/#commitbody) configuration seems like it is designed to be able to specify commit trailers in it since renovate even has a [gitSignOff](https://docs.renovatebot.com/presets-default/#gitsignoff) preset which should add the `"Signed-off-by: {{{gitAuthor}}}"` trailer.

Git trailers follow specific requirements and need to be the last block of a commit message for them to actually be trailers. https://git-scm.com/docs/git-interpret-trailers

Renovate appears to properly handle this for all cases except for gerrit since it needs to insert some other trailers
https://github.com/renovatebot/renovate/blob/8d31e5cd6222c01b02f4acfe8a97a830dabbfef8/lib/modules/platform/gerrit/scm.ts#L118-L121
Renovate adds these internal trailers as the last entry to `commit.message` array which simple-git then converts to a single string by joining them with two line breaks between them.

This ultimately means that renovate can't be configured to add on additional git trailers when using gerrit.

---
Example:

Renovate config
```
{
  "commitBody": "My-Trailer: test"
}
```

Ultimate gerrit commit message:
```
chore(deps): update mydep digest to v1.0.1

My-Trailer: test

Renovate-Branch: my-branch-name
Change-Id: I24e1e7c57c808ba81de51b118e3146336f771bba
```

In the commit message above, we can see that `My-Trailer` will not be interpreted as a trailer because it is not in the last group of lines.

---

One potential fix might be to append these renovate generated trailers to the last element of `message` if `message` has more than one element.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by felipecrs** [comment, marked as answer]
I proposed a PR that "fixes" this:

- https://github.com/renovatebot/renovate/pull/44651
